lib/core/src/getRodsEnv.cpp

```diff
@@ -33,13 +33,112 @@
+        copy_int(irods::KW_CFG_ENCRYPTION_SALT_SIZE, _env.rodsEncryptionSaltSize);
+
+        copy_string(irods::KW_CFG_DEFAULT_HASH_SCHEME, _env.rodsDefaultHashScheme);
+        copy_string(irods::KW_CFG_MATCH_HASH_POLICY, _env.rodsMatchHashPolicy);
+
+        copy_string(irods::KW_CFG_DEFAULT_RESOURCE_NAME, _env.rodsDefResource);
+        copy_int(irods::KW_CFG_CONNECTION_POOL_REFRESH_TIME, _env.irodsConnectionPoolRefreshTime);
+
+        if (const auto tls_iter = config.find(irods::KW_CFG_TLS_CLIENT); tls_iter != std::end(config)) {
+            // clang-format off
+            const auto copy_tls_string = []<std::size_t N>(char (&_dst)[N], const nlohmann::json& _src) {
+                _src.get_ref<const std::string&>().copy(_dst, N - 1);
+            };
+            // clang-format on
+
+            for (auto&& [k, v] : tls_iter->items()) {
+                if (irods::KW_CFG_TLS_CA_CERTIFICATE_FILE == k) {
+                    copy_tls_string(_env.irodsSSLCACertificateFile, v);
+                }
+                else if (irods::KW_CFG_TLS_CA_CERTIFICATE_PATH == k) {
+                    copy_tls_string(_env.irodsSSLCACertificatePath, v);
+                }
+                else if (irods::KW_CFG_TLS_VERIFY_SERVER == k) {
+                    copy_tls_string(_env.irodsSSLVerifyServer, v);
+                }
+            }
+        }
+
+        // If the configuration is not set for the TCP keepalive options, set the value to something invalid. This
+        // indicates that we should not set the option on the socket, which will allow the socket to use the kernel
+        // configuration.
+        _env.tcp_keepalive_intvl = config.value(irods::KW_CFG_TCP_KEEPALIVE_INTVL_IN_SECONDS, -1);
+        if (_env.tcp_keepalive_intvl < 0) {
+            _env.tcp_keepalive_intvl = -1;
+        }
+
+        _env.tcp_keepalive_probes = config.value(irods::KW_CFG_TCP_KEEPALIVE_PROBES, -1);
+        if (_env.tcp_keepalive_probes < 0) {
+            _env.tcp_keepalive_probes = -1;
+        }
+
+        _env.tcp_keepalive_time = config.value(irods::KW_CFG_TCP_KEEPALIVE_TIME_IN_SECONDS, -1);
+        if (_env.tcp_keepalive_time < 0) {
+            _env.tcp_keepalive_time = -1;
+        }
+
+        const auto& advanced_settings = config.at(irods::KW_CFG_ADVANCED_SETTINGS);
+        _env.irodsDefaultNumberTransferThreads =
+            advanced_settings.at(irods::KW_CFG_DEF_NUMBER_TRANSFER_THREADS).get<int>();
+        _env.irodsMaxSizeForSingleBuffer = advanced_settings.at(irods::KW_CFG_MAX_SIZE_FOR_SINGLE_BUFFER).get<int>();
+        _env.irodsTransBufferSizeForParaTrans =
+            advanced_settings.at(irods::KW_CFG_TRANS_BUFFER_SIZE_FOR_PARA_TRANS).get<int>();
+
+        if (auto iter = config.find(irods::KW_CFG_PLUGIN_DIRECTORY); iter != std::end(config)) {
+            iter->get_ref<const std::string&>().copy(
+                _env.irodsPluginDirectory, sizeof(rodsEnv::irodsPluginDirectory) - 1);
+        }
```

Comment:
These lines need to be moved into commit 5922eb45f44a4b35a883646f0376d0379ded3b42.

Reply:
Done.